plugins/modules/aci_l3out.py

```diff
@@ -394,26 +408,20 @@ def main():
 
     if l3protocol:
         if "eigrp" in l3protocol and asn is None:
-            module.fail_json(
-                msg="Parameter 'asn' is required when l3protocol is 'eigrp'"
-            )
+            module.fail_json(msg="Parameter 'asn' is required when l3protocol is 'eigrp'")
         if "eigrp" not in l3protocol and asn is not None:
-            module.warn(
-                "Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored"
-            )
+            module.warn("Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored")
+        if "ospf" in l3protocol and ospf is None:
+            module.fail_json(msg="ospf specifications are required when l3protocol is 'ospf'")
+        if "ospf" not in l3protocol and ospf is not None:
+            module.warn("Parameter 'ospf' is only applicable when l3protocol is 'ospf'. The OPSF specifications will be ignored")
```

Comment:
Do we test the warning scenario flows?

Reply:
Did you check for a way to do this?

Reply:
I asked Shreyas but he told there is no way to do it or maybe there is way he is not aware of.

Reply:
@gmicol The warn messages invoked in the module are appended to a list called warnings. If you want to test this scenario, can you try the following and let me know if it works:

```
- name: Add a new L3Out
  cisco.aci.aci_l3out:
    host: apic
    username: admin
    password: SomeSecretPassword
    tenant: production
    name: prod_l3out
    description: L3Out for Production tenant
    domain: l3dom_prod
    vrf: prod
    l3protocol: bgp
    ospf:
      area_cost: 1
      area_ctrl: [ summary, redistribute ]
      area_id: 0.0.0.1
      area_type: regular
      multipod_internal: no
    state: present
  register: warn_ospf

- name: check warning for ospf
      assert: 
        that: warn_ospf.warnings.0 == "Parameter 'ospf' is only applicable when l3protocol is 'ospf'. The OPSF specifications will be ignored"
```

Reply:
it did work! thanks @shrsr